runtime/reconcilers/model.go

```diff
@@ -644,6 +659,94 @@ func (r *ModelReconciler) acquireExecutor(ctx context.Context, opts *drivers.Mod
 	return executorName, e, release, nil
 }
 
+func (r *ModelReconciler) acquireExecutorWithStage(ctx context.Context, opts *drivers.ModelExecutorOptions) (string, drivers.ModelExecutor, func(), error) {
```

Comment:
Since this function actually does execution, not just acquisition (which sounds "light"), may be worth refactoring the name

Reply:
Okay slightly refactored the code so that we can call this `executeWithStage`